docs/guides/distribution/updater.md

```diff
@@ -224,113 +132,103 @@ The alternate update technique uses a plain JSON file, storing your update metad
 
-```
-target/release/bundle
-└── macos
-    └── app.app
-    └── app.app.tar.gz (update bundle)
-    └── app.app.tar.gz.sig
-```
+By default, the updater uses a built-in dialog API from Tauri. The dialog will only check for a new update when the app was just launched or when you manually [emit](../../api/js/event.md#emit) the `"tauri://update"` event.
 
-### Windows
+![New Update](https://i.imgur.com/UMilB5A.png)
 
-On Windows, we create a .zip from the MSI; when downloaded and validated, we run the MSI install.
+The dialog release notes are represented by the update `notes` provided by the [server](#server-support).
+If the user accepts, the update is downloaded and installed. Afterwards, the user is prompted to restart the application.
 
-```
-target/release/bundle
-└── msi
-    └── app.x64.msi
-    └── app.x64.msi.zip (update bundle)
-    └── app.x64.msi.zip.sig
-```
+### Custom Dialog and Logic
 
-### Linux
+:::caution
 
-On Linux, we create a .tar.gz from the AppImage.
+You need to disable the built-in dialog in your [Tauri configuration](#tauri-configuration) to enable the JavaScript APIs and updater events!
 
-```
-target/release/bundle
-└── appimage
-    └── app.AppImage
-    └── app.AppImage.tar.gz (update bundle)
-    └── app.AppImage.tar.gz.sig
-```
+:::
 
-## Signing updates
+#### Rust
 
-We offer a built-in signature to ensure your update is safe to be installed.
+Please see the updater module documentation at [docs.rs] for the Rust API.
 
-To sign your updates, you need two things.
+#### JavaScript
 
-The _Public-key_ (pubkey) should be added inside your `tauri.conf.json` to validate the update archive before installing.
+For the complete API docs see [here](../../api/js/updater.md). An example using the JS APIs could look like this:
 
-The _Private key_ (privkey) is used to sign your update and should NEVER be shared with anyone. Also, if you lost this key, you'll NOT be able to publish a new update to the current user base. It's crucial to save it in a safe place, and you can always access it.
+```js title=updater.ts
+import {
+  checkUpdate,
+  installUpdate,
+  onUpdaterEvent,
+} from '@tauri-apps/api/updater'
+import { relaunch } from '@tauri-apps/api/process'
 
-To generate your keys, you need to use the Tauri CLI:
+const unlisten = await onUpdaterEvent(({ error, status }) => {
+  // This will log all updater events, including status updates and errors.
+  console.log('Updater event', error, status)
+})
 
-```shell
-tauri signer generate -w ~/.tauri/myapp.key
-```
+try {
+  const { shouldUpdate, manifest } = await checkUpdate()
 
-You have multiple options available
+  if (shouldUpdate) {
+    // You could show a dialog asking the user if they want to install the update here.
+    console.log(
+      `Installing update ${manifest?.version}, ${manifest?.date}, ${manifest?.body}`
+    )
 
-```
-Generate keypair to sign files
+    // Install the update. This will also restart the app on Windows!
+    await installUpdate()
```

Comment:
Technical question, but why is this different behaviour on platforms? If there's no reason then I think this should be opened as an issue in Tauri

Reply:
On windows the app needs to be closed to be able to install the new version with the MSI installer. I'm not sure if that can be worked around.

Reply:
yeah, there are a lot of quirks in the updater that need a bit refinement.